### PR TITLE
[REVIEW] Dropping `find_first_value` and `find_last_value`

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
Removed `find_first_value` and `find_last_value`. There is already `find_first_value` and `find_last_value` in numerical column which uses numba kernel, so the additional change that has been added to `column.py` will act as backup.

#4945 triggered this removal.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -616,22 +616,27 @@ def find_first_value(self, value):
...
-        indices = np.argwhere(arr == value)
+        from cudf.core.index import RangeIndex
+
+        # FIXME: Inefficient, may be need a libcudf api
+        index = RangeIndex(0, stop=len(self))
+        indices = index.take(self == value)
         if not len(indices):
             raise ValueError("value not found")
-        return indices[-1, 0]
+        return indices[0]
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/column.py:626]: It really stinks to have to materialize a 0 --> N column here just to get the first `True` value from the elementwise equals op. Could we use an `argmax` instead here? Something like:
```
return (self == value).argmax()
```
That should return the idx of the first True value which can be used to grab the corresponding index which could be used to grab a scalar from the Index column if needed.

Comment (kkraus14) [reply python/cudf/cudf/core/column/column.py:626]: I imagine there may be a better libcudf function already available to accomplish this though.

Comment (rgsl888prabhu) [reply python/cudf/cudf/core/column/column.py:626]: Don't think we have any such API suport in lubcudf.

File: python/cudf/cudf/core/column/column.py [hunk 2]
```diff
@@ -616,22 +616,27 @@ def find_first_value(self, value):
         """
         Returns offset of first value that matches
         """
-        # FIXME: Inefficient find in CPU code
-        arr = self.to_array()
-        indices = np.argwhere(arr == value)
+        from cudf.core.index import RangeIndex
+
+        # FIXME: Inefficient, may be need a libcudf api
+        index = RangeIndex(0, stop=len(self))
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/column.py:622]: We found doing these imports can actually get somewhat expensive.

```suggestion
        # FIXME: Inefficient, may be need a libcudf api
        index = cudf.core.index.RangeIndex(0, stop=len(self))
```

File: python/cudf/cudf/core/column/column.py [hunk 3]
```diff
@@ -616,22 +616,27 @@ def find_first_value(self, value):
...
     def find_last_value(self, value):
         """
         Returns offset of last value that matches
         """
-        arr = self.to_array()
-        indices = np.argwhere(arr == value)
+        from cudf.core.index import RangeIndex
+
+        # FIXME: Inefficient, may be need a libcudf api
+        index = RangeIndex(0, stop=len(self))
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/column.py:635]:
```suggestion
        # FIXME: Inefficient, may be need a libcudf api
        index = cudf.core.index.RangeIndex(0, stop=len(self))
```

File: python/cudf/cudf/core/column/column.py (hunk 2 above)
Comment (kkraus14) [reply python/cudf/cudf/core/column/column.py:622]: This wasn't applied as far as I can tell.

File: python/cudf/cudf/core/column/column.py (hunk 3 above)
Comment (kkraus14) [reply python/cudf/cudf/core/column/column.py:635]: This wasn't addressed either as far as I can tell.

**Comment by rgsl888prabhu** [comment]
Needs some changes, will be pushing an update tomorrow.

**Comment by kkraus14** [comment]
Thanks for the heads up @rgsl888prabhu

**Comment by kkraus14** [comment]
@rgsl888prabhu Seems like @shwina jumped the gun here so just add any additional changes as a followup PR 😄

**Comment by rgsl888prabhu** [comment]
I was planning same, no issues :)

**Pull request merged**